SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -269,6 +270,50 @@ void USpatialPackageMapClient::ClearRemovedDynamicSubobjectObjectRefs(const Work
 	}
 }
 
+void USpatialPackageMapClient::RemoveBNetLoadOnClientRuntimeRemovedComponents(const Worker_EntityId& EntityId,
+																			  const TArray<SpatialGDK::ComponentData>& NewComponents,
+																			  const SpatialGDK::ActorSystem& ActorSystem
+
+)
+{
+	auto ContainedInComponentsArr = [this, &NewComponents, &EntityId](const FUnrealObjectRef CheckComponentObjRef) {
+		for (const SpatialGDK::ComponentData& Component : NewComponents)
+		{
+			uint32 Offset = 0;
+			NetDriver->ClassInfoManager->GetOffsetByComponentId(Component.GetComponentId(), Offset);
+
+			if (FUnrealObjectRef(EntityId, Offset) == CheckComponentObjRef)
+			{
+				return true;
+			}
+		}
+		return false;
+	};
+
+	// Go over each stored sub-object and determine whether it is contained within the new components array
+	// If it is not contained within the new components array, it means the sub-object was removed while out of the client's interest
+	// If so, remove it now
+	for (auto DynamicSubObjectIterator = RemovedDynamicSubobjectObjectRefs.CreateIterator(); DynamicSubObjectIterator;
+		 ++DynamicSubObjectIterator)
+	{
+		if (DynamicSubObjectIterator->Key.Entity == EntityId)
+		{
+			if (!ContainedInComponentsArr(DynamicSubObjectIterator->Key))
+			{
+				if (UObject* Object = NetDriver->PackageMap->GetObjectFromNetGUID(DynamicSubObjectIterator->Value, false))
+				{
+					UE_LOG(LogSpatialPackageMap, Verbose,
+						   TEXT("A SubObject (ObjectRef offset: %u) on bNetLoadOnClient actor with entityId %d was destroyed while the "
+								"actor was out of the client's interest. Destroying the SubObject now."),
+						   DynamicSubObjectIterator->Key.Offset, EntityId);
+					ActorSystem.DestroySubObject(EntityId, Object, DynamicSubObjectIterator->Key);
```

Comment:
Hi @jacquese, from blame I see you've written a lot of the actor system file - do you have any thoughts on this specific part of the PR? The context is we need to destroy a subobject, based on some information we've stored in SpatialPackageMapClient. What I've got here is a direct call to an extracted destroysubobject method that does this. Though with this destroysubobject needs to be a public method which is slightly messy.

An alternative to this would be to return a list containing pairs of object and objectref - do you have a preference?

Reply:
Another alternative may be to move this function to the actorsystem instead
Not sure how much we would need to expose from the packagemap in that case

Reply:
AFAICT, this should be safe (and probably nicer) to move to the actor system. `RemovedDynamicSubobjectObjectRefs` doesn't seem to belong in the package map client anymore, and its location appears to be arbitrary. I would recommend moving it as a whole to the actor system for now.